test/com/google/javascript/jscomp/TypeInferenceTest.java

```diff
@@ -3798,6 +3806,25 @@ public void testDynamicImport2() {
             "Promise<{\n" + "  Bar: function(): string,\n" + "  default: number\n" + "}>");
   }
 
+  @Test
+  public void testDynamicImportAfterModuleRewriting() {
+    compiler.getOptions().setProcessCommonJSModules(true);
+    withModules(
+        ImmutableList.of(
+            lines(
+                "const module$testcode0 = {};",
+                "/** @const */ module$testcode0.default = 1;",
+                "/** @return {string} */ function Bar() { return 'bar'; };",
+                "/** @const */ module$testcode0.Bar = Bar;"),
+            // modules are named of the format `testcode#` based off their index.
+            // testcode0 refers the first module.
+            "const foo = import('./testcode0');"));
+
+    assertType(getType("foo"))
+        .toStringIsEqualTo(
+            "Promise<{\n" + "  Bar: function(): string,\n" + "  default: number\n" + "}>");
```

Comment:
could you use `lines()` and a trailing `//` comment on the first argument to make these more readable?

Reply:
fixed